### DISCUSSION TITLE
Disabling Peer 2 Peer Mode

### DISCUSSION THREAD

**Discussion by t3chn0m4g3** [SentryPeer/SentryPeer#77] [Q&A] [answered]
With P2P mode defaulting to enabled, is there an option beside `-b 127.0.0.1` to deactivate P2P mode?

**Comment by ghenry** [comment, marked as answer]
Hi @t3chn0m4g3 

What version are you using? `-p` or `SENTRYPEER_PEER_TO_PEER` is needed to enable p2p mode. It's off by default :-)

That's been there since [v1.4.0](https://github.com/SentryPeer/SentryPeer/releases/tag/v1.4.0)

[v2.0.0](https://github.com/SentryPeer/SentryPeer/releases/tag/v2.0.0) is recommended at the moment. `-b` was added in [v1.4.1](https://github.com/SentryPeer/SentryPeer/releases/tag/v1.4.1), so you must be on that version?

Thanks!

**Reply by t3chn0m4g3** [reply]
@ghenry I am on the Alpine package version `v2.0.0`. The only options I used are `-jar` and yet ...
```
sentrypeer    | SentryPeer db file location set via cli -f to: /var/log/sentrypeer/sentrypeer.db
sentrypeer    | SentryPeer json log file location set via cli -f to: /var/log/sentrypeer/sentrypeer.json
sentrypeer    | SentryPeer node id: 00f1fda5-d6e4-47e5-8998-66096f46ef1a
sentrypeer    | Starting sentrypeer...
sentrypeer    | API mode enabled, starting http daemon...
sentrypeer    | SIP mode enabled...
sentrypeer    | Peer to Peer DHT mode enabled...
sentrypeer    | Starting peer to peer DHT mode using OpenDHT-C lib version '2.4.4'...
sentrypeer    | Configuring local address...
sentrypeer    | Creating sockets...
sentrypeer    | Binding sockets to local address...
sentrypeer    | Listening for incoming UDP connections...
sentrypeer    | SIP responsive mode enabled. Will reply to SIP probes...
sentrypeer    | Listening for incoming TCP connections...
sentrypeer    | Peer to peer DHT mode started.
sentrypeer    | DHT InfoHash for key 'bad_actors' is: 14d30143330e2e0e922ed4028a60ff96a59800ad
sentrypeer    | Bootstrapping the DHT
sentrypeer    | Waiting 5 seconds for bootstrapping to bootstrap.sentrypeer.org...
sentrypeer    | Listening for changes to the bad_actors DHT key
```
Am I missing something?
Even setting `ENV SENTRYPEER_PEER_TO_PEER=0` keeps it enabled.

Running with `/usr/bin/sentrypeer -b 127.0.0.1 -jar -f /var/log/sentrypeer/sentrypeer.db -l /var/log/sentrypeer/sentrypeer.json` keeps me from receiving P2P messages 🙃

**Reply by ghenry** [reply]
How strange. Let me investigate. Thanks!

**Reply by t3chn0m4g3** [reply]
Thanks!

**Reply by ghenry** [reply]
Are you setting `ENV SENTRYPEER_PEER_TO_PEER=0`

If so, we don't care WHAT `ENV SENTRYPEER_PEER_TO_PEER` is, only IF it's present.

**Reply by ghenry** [reply]
This is what you should see with debug enabled:

```
localhost ~/src/sentrypeer [main*]$ ./sentrypeer -V
This is sentrypeer version: 2.0.1, git rev: e82f4fa

localhost ~/src/sentrypeer [main*]$ ./sentrypeer -d -jar -f /home/ghenry/src/sentrypeer/sentrypeer.db -l /home/ghenry/src/sentrypeer/sentrypeer.json
SentryPeer db file location set via cli -f to: /home/ghenry/src/sentrypeer/sentrypeer.db
SentryPeer json log file location set via cli -f to: /home/ghenry/src/sentrypeer/sentrypeer.json
SentryPeer node id: 0523c46a-036f-4a8d-a08f-d115406943d6
Starting sentrypeer...
API mode enabled, starting http daemon...
SIP mode enabled...
Configuring local address...
Creating sockets...
Binding sockets to local address...
Listening for incoming UDP connections...
SIP responsive mode enabled. Will reply to SIP probes...
Listening for incoming TCP connections...
^C
Stopping sentrypeer...
Stopping http daemon...
Stopping sip daemon...
Stopped sentrypeer
```

**Reply by ghenry** [reply]
If I do this, I get what you get:

```
SENTRYPEER_PEER_TO_PEER=0 ./sentrypeer -d -jar -f /home/ghenry/src/sentrypeer/sentrypeer.db -l /home/ghenry/src/sentrypeer/sentrypeer.json
SentryPeer db file location set via cli -f to: /home/ghenry/src/sentrypeer/sentrypeer.db
SentryPeer json log file location set via cli -f to: /home/ghenry/src/sentrypeer/sentrypeer.json
SentryPeer node id: d634941a-831a-4037-adc1-89cb648fff73
Starting sentrypeer...
API mode enabled, starting http daemon...
SIP mode enabled...
Peer to Peer DHT mode enabled...
Starting peer to peer DHT mode using OpenDHT-C lib version '2.4.10'...
Configuring local address...
Creating sockets...
Binding sockets to local address...
Listening for incoming UDP connections...
SIP responsive mode enabled. Will reply to SIP probes...
Listening for incoming TCP connections...
Peer to peer DHT mode started.
DHT InfoHash for key 'bad_actors' is: 14d30143330e2e0e922ed4028a60ff96a59800ad
Bootstrapping the DHT
Waiting 5 seconds for bootstrapping to bootstrap.sentrypeer.org...
^C
Listening for changes to the bad_actors DHT key
Stopping sentrypeer...
Stopping http daemon...
Stopping sip daemon...
Stopping peer to peer DHT daemon...
Stopped sentrypeer
```

I suppose this is a user surprise/unexpected behaviour as the docs state `SENTRYPEER_PEER_TO_PEER=1` to enable. You'd _presume_ `=0` disables, but disabled is the default. I'll clean up the docs as it does say p2p is default on. The CLI doesn't state that it's off by default.

How would you like the current defaults stated as a user?

**Reply by t3chn0m4g3** [reply]
Thanks, will rebuild this to your feedback and update this thread afterwards.

**Reply by t3chn0m4g3** [reply]
Confirmed, this works as pointed out.

Thank you for your help!

**Reply by ghenry** [reply]
Would you like SENTRYPEER_PEER_TO_PEER and other env vars to understand that =0 means to disable that feature or have the env var not there at all as the preferred way?

**Reply by t3chn0m4g3** [reply]
Thank you for asking, yes I think having ENVs set to `=0` to disable settings would be a handy feature, especially WRT to containers.

**Reply by ghenry** [reply]
It's tricky because the only thing on by default is the "comsume a SIP packet" feature. Everything else you need to enable via CLI or set the ENV. I was careful to not make surprises, or at least I thought I was :-)

The next time I add a default on feature, I'll make =0 work. That fair?

**Reply by t3chn0m4g3** [reply]
I see what you mean, this might cause unexpected conditions, so I think removing the ENVs is probably the better choice for the long run.

**Reply by ghenry** [reply]
Will have a think and leave this here for others to read.
